README.md

```diff
@@ -21,147 +24,131 @@ or to use in an npm script
-2017-07-12T18:56:39.199Z INFO done
-```
+Along with the command line, there is also a public API.  The API mirrors the commands.
 
-If you login to your local Openshift cluster and navigate to the "node-demo" project, there should be a new deployment with a route to your application.
+* resource
 
-For example: `http://nodejs-rest-http-node-demo.192.168.99.100.nip.io/`
+* applyResource
 
-_please note: Currently, once a route, service, deployment config, build config, and imagestream config are created, those are re-used. The only thing that changes from deployment to deployment is the source code.  This will change in the future_
+* build
 
-### .Nodeshift Directory
+* deploy
 
-The `.nodeshift` directory is responsible for holding your resource files.  These are `.yml` files that describe you service, deployments and routes.
+* undeploy
 
-Currently, nodeshift will only create resources based on the files specified,  in the future, its possible somethings could be created by default
+Options that you can specify on the command line, can also be passed as an options object to the API
 
-#### Commands & Goals
+#### Example Usage
 
-By default, if you run just `nodeshift`, it will run the `deploy` goal, which runs everything.
+    const nodeshift = require('nodeshift');
 
-**resource** - will parse and create the application resources files on disk
+    // Deploy an Application
+    nodeshift.deploy().then((response) => {
+        console.log(response);
+        console.log('Application Deployed')
+    }).catch((err) => {
+        console.log(err);
+    })
 
-**apply-resource** - does the resource goal and will also push that to your running cluster
+_please note: Currently, once a route, service, deployment config, build config, and imagestream config are created, those are re-used. The only thing that changes from deployment to deployment is the source code.  For application resources, you can update them by undeploying and then deploying again.  BuildConfigs and Imagestreams can be re-created using the --build.recreate flag_
 
-**deploy** - does everything, the archiving of code and everything that apply-resource does
+## Advanced Options
 
-**undeploy** - removes things that were deployed with the resource command
+While nodeshift tries to be very opinionated during the deployment process, but there are options available to pass to the cli or the API.
 
-**build** - archives the code, creates build config and imagestream and pushes the binary to the cluster
+#### version
+Outputs the current version of nodeshift
 
-#### Template Parameters
+#### projectLocation
+Changes the default location of where to look for your project. Defaults to your current working directory(CWD)
 
-Some templates might need to have a value set at "run time".  For example, in the template below, we have the `${SSO_AUTH_SERVER_URL}` parameter:
+#### configLocation
+Changes the default location of where to look for a configfile. Defaults to ~/.kube/
```

Comment:
s/of where to look for a configfile/of the config file/

Reply:
What goes in the config file?

Reply:
tbh,  i don't know if we should really even let the user do this.  All that config stuff is created and managed by the oc commands(and kubectl also),

Reply:
I agree. I don't think it is something that needs to be exposed.

Reply:
ok,  removing this

Reply:
I went to upgrade to latest nodeshift in [a project that runs in a Katacoda environment](https://learn.openshift.com/middleware/rhoar-getting-started-nodejs/), where the configuration file for the `oc` command is _not_ at `~/.kube/config`. It's actually at `/openshift.local.config/master/admin.kubeconfig`, and I was using the `configLocation` in nodeshift to point at it. I'm not an expert at kubernetes config locations and don't know if this is a bad practice, but removing this prevents my project from using latest nodeshift...